javascript/react/src/components/Codebook/CodebookFieldEditor.tsx

```diff
@@ -0,0 +1,149 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { DataType, VariableNature } from '@datashaper/schema'
+import { EnumDropdown } from '@essex/components'
+import { TextField } from '@fluentui/react'
+import { memo, useCallback } from 'react'
+import { If, Then } from 'react-if'
+
+import {
+	Container,
+	FieldContainer,
+	OutsideLabel,
+} from './CodebookFieldEditor.styles.js'
+import type { CodebookFieldEditorProps } from './CodebookFieldEditor.types.js'
+import {
+	CodebookFields,
+	DEFAULT_CODEBOOK_FIELDS,
+} from './CodebookFieldEditor.types.js'
+import { MappingFields } from './MappingFields.js'
+import { StatsField } from './StatsField.js'
+
+export const CodebookFieldEditor: React.FC<CodebookFieldEditorProps> = memo(
+	function CodebookFieldEditor({
+		field,
+		onChange,
+		tableView,
+		hideLabel,
+		visibleFields = DEFAULT_CODEBOOK_FIELDS,
+	}) {
+		const visibleFieldsSet = new Set<CodebookFields>()
+		visibleFields.forEach(unit => {
+			visibleFieldsSet.add(unit)
+		})
+
+		const onChangeField = useCallback(
+			(val: any) => {
+				onChange({ ...field, ...val })
+			},
+			[field, onChange],
+		)
+
+		return (
+			<Container className="codebook-column">
+				<StatsField
+					tableView={tableView}
+					onChange={onChange}
+					field={field}
+				></StatsField>
+				<If condition={visibleFieldsSet.has(CodebookFields.DisplayName)}>
+					<Then>
+						{tableView && !hideLabel && <OutsideLabel>Display</OutsideLabel>}
+						<FieldContainer className="field">
+							<TextField
+								label={!tableView && !hideLabel ? 'Display' : undefined}
+								borderless
+								disabled={field.exclude}
+								name="displayName"
+								value={field.title}
+								onChange={(_, val) => onChangeField({ title: val })}
+							/>
+						</FieldContainer>
+					</Then>
+				</If>
+				<If condition={visibleFieldsSet.has(CodebookFields.Description)}>
+					<Then>
+						{tableView && !hideLabel && (
+							<OutsideLabel>Description</OutsideLabel>
+						)}
+						<FieldContainer className="field">
+							<TextField
+								label={!tableView && !hideLabel ? 'Description' : undefined}
+								borderless
+								multiline
+								resizable={false}
+								disabled={field.exclude}
```

Comment:
This disabled doesn't seem to cover everything, so only part of the column goes grey. You may want to make even the stats header look disabled, and definitely make sure the mapping cell becomes disabled. It should look clear that the entire column is off, and the only way to get it back on is to re-check the box.